File: touralbum/app/src/main/java/com/example/touralbum/ui/reminder/util/DateTimeUtil.java
package com.example.touralbum.ui.reminder.util;

import com.example.touralbum.ui.reminder.Constants;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;



public class DateTimeUtil {
    public static String dateToStr(Date date) {
        SimpleDateFormat sdf = new SimpleDateFormat(Constants.DEFAULT_TIME_FORMAT, Locale.CHINA);
        return sdf.format(date);
    }

    public static Date str2Date(String src)  {
        SimpleDateFormat sdf = new SimpleDateFormat(Constants.DEFAULT_TIME_FORMAT, Locale.CHINA);
        try {
            return sdf.parse(src);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return null;
    }

    public static boolean needClock(Date date, Date date2) {
       if (Math.abs(((double) (date.getTime() - date2.getTime())) / 1000) <= 1.00001) {
           return true;
       }
       return false;
    }

    public static boolean canClock(Date date) {
        if (Math.abs(date.getTime() - new Date().getTime()) < 5000) {
            return true;
        }
        return false;
    }
}

File: touralbum/app/src/main/java/com/example/touralbum/ui/reminder/util/DateTimeUtil.kt
package com.example.touralbum.ui.reminder.util

import com.example.touralbum.ui.reminder.Constants
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

/**
 * @author jsbintask@gmail.com
 * @date 2018/4/25 15:53
 */
object DateTimeUtil {
    fun dateToStr(date: Date?): String {
        val sdf = SimpleDateFormat(Constants.DEFAULT_TIME_FORMAT, Locale.CHINA)
        return sdf.format(date)
    }

    fun str2Date(src: String?): Date? {
        val sdf = SimpleDateFormat(Constants.DEFAULT_TIME_FORMAT, Locale.CHINA)
        try {
            return sdf.parse(src)
        } catch (e: ParseException) {
            e.printStackTrace()
        }
        return null
    }

    fun needClock(date: Date, date2: Date): Boolean {
        return if (Math.abs((date.time - date2.time).toDouble() / 1000) <= 1.00001) {
            true
        } else false
    }

    fun canClock(date: Date): Boolean {
        return if (Math.abs(date.time - Date().time) < 5000) {
            true
        } else false
    }
}